src/sentry/grouping/strategies/message.py

```diff
@@ -113,10 +146,53 @@
         =false
     )
 """
-)
+
+_parameterization_regex = re.compile(_parameterization_regex_str)
+
+
+@dataclasses.dataclass()
+class ParameterizationExperiment:
+    name: str
+    regex: Any  # re.compile(...)
+    counter: int = 0
+
+
+# Note that experiments are run AFTER the initial replacements. Which means they MUST not catch replacements made
+# in the primary parameterization regex. E.g. "md5" might be caught by the "uniq_id" experiment, so it is explicitly excluded
+_parameterization_regex_experiments = [
+    ParameterizationExperiment(
+        name="uniq_id",
+        regex=re.compile(
+            _parameterization_regex_str
+            + r"""|
+                (?P<uniq_id>
+                    \b
+                    (?!(md5|sha1)) # TODO: remove this line after experiment is done
+                    (?!\w*?[\.:\-]\w*?\b) # No colons, dots, or dashes
+                    (?=\w*?[0-9]\w*?\b) # At least one digit
+                    (?=\w*?[a-zA-Z]\w*?\b) # At least one letter
+                    [\w_]+?
+                    \b
```

Comment:
This will also much things like for example `python3, cloud9, 1password`.

Reply:
is this an argument for 2 digits? That's more conservative and probably safer

Reply:
I think we can get some data by turning it on quickly and see how it performs

Reply:
Rather than having minimum number of digits or letters, I think it has to make sure they're interleaved 

so `ab1c23d` -> yes
but `abcd123` or `123abcd` -> no

so I guess negative lookback for all letters followed by all digits, and all digits followed by all letters.